dali/operators/expressions/arithmetic.cc

```diff
@@ -21,18 +21,19 @@ namespace dali {
 
 template <>
 void ArithmeticGenericOp<CPUBackend>::RunImpl(HostWorkspace &ws) {
+  PrepareTilesForTasks(ws);
```

Comment:
It is not very intuitive that `exec_order_` is set by `PrepareTilesForTasks`

Reply:
I moved them out but still access the output vector by reference instead of returning it to reduce the reallocations if possible.